src/backend/distributed/utils/metadata_cache.c

```diff
@@ -636,11 +840,19 @@ ResetDistTableCacheEntry(DistTableCacheEntry *cacheEntry)
 					pfree(DatumGetPointer(shardInterval->maxValue));
 				}
 			}
+
+			pfree(shardInterval);
 		}
 
-		pfree(cacheEntry->shardIntervalArray);
-		cacheEntry->shardIntervalArray = NULL;
+		pfree(cacheEntry->sortedShardIntervalArray);
+		cacheEntry->sortedShardIntervalArray = NULL;
 		cacheEntry->shardIntervalArrayLength = 0;
+
+		cacheEntry->hasUninitializedShardInterval = false;
+		cacheEntry->hasUniformHashDistribution = false;
+
+		pfree(cacheEntry->shardIntervalCompareFunction);
+		cacheEntry->shardIntervalCompareFunction = NULL;
```

Comment:
One question here, do I need to free hashFunction as well? I think I shouldn't be doing it since we're using typecache enrty. Is that right?